### DISCUSSION TITLE
Catalog Request not displaying any Contract Offers

### DISCUSSION THREAD

**Discussion by pablobiedma** [eclipse-edc/Samples#18] [Q&A] [answered]
Dear all, I am a beginner using the connector and playing around with the 'Samples' tutorial. I would really appreciate your help with something: 

I am trying to display all available Contract Offers. After reading the swagger API endpoint documentation, I thought I should do a request like this: 

`curl -X POST -H "Content-Type: application/json" -H "X-Api-Key: password" -d @transfer/transfer-04-open-telemetry/contractdefinition.json http://localhost:9192/api/v1/management/catalog/request`

This is what the body of the request contains in the contractdefinition.json file: 

```
{
  "providerUrl": "http://localhost:9292/api/v1/ids/data",
  "querySpec": {
    "filter": null,
    "limit": 50,
    "offset": 0,
    "sortField": null,
    "sortOrder": "ASC"
  }
}
```

I would expect an output similar to this: 

`{"id":"default","contractOffers":[{"id":"1:d2e005c1-826f-43a8-889b-abc625f68c5a","policy":{"permissions":[{"edctype":"dataspaceconnector:permission","uid":null,"target":"test-document","action":{"type":"USE","includedIn":null,"constraint":null},"assignee":null,"assigner":null,"constraints":[],"duties":[]}],"prohibitions":[],"obligations":[],"extensibleProperties":{},"inheritsFrom":null,"assigner":null,"assignee":null,"target":"test-document","@type":{"@policytype":"set"}},"asset":{"id":"test-document","createdAt":1674635154040,"properties":{"asset:prop:byteSize":null,"asset:prop:id":"test-document","asset:prop:fileName":null}},"provider":"urn:connector:provider","consumer":"urn:connector:consumer","offerStart":null,"offerEnd":null,"contractStart":"2023-01-25T09:25:54.002+01:00","contractEnd":"2023-01-25T09:25:54.002Z"}]}`

However I am getting this output as if there were no offers available: 

`{"id":"default","contractOffers":[]}%
`

I believe the error might be in the contractdefinition.json file. 
Does any of you know how such a request can be done properly? 

Thank you so much for your help.

**Comment by ronjaquensel** [comment, marked as answer]
As providerUrl in the request you are stating `http://localhost:9292/api/v1/ids/data`. If you did not make any changes to the port configurations, the **consumer** is running on 9292, which doesn't offer any data. The provider is running on port 8282, so to query the provider's catalog you would need to set `"providerUrl": "http://localhost:8282/api/v1/ids/data"`.
